packages/core/svelte/src/walletStore.ts

```diff
@@ -375,3 +371,8 @@ export function destroyAdapter() {
     adapter.off('disconnect', onDisconnect);
     adapter.off('error', onError);
 }
+
+if (typeof window !== 'undefined') {
```

Comment:
I don't think this is needed. I'll be surprised if the browser doesn't clean these up automatically

Reply:
In this case the issue is timing and race condition prevention. When the window unloads, the adapter's `disconnect` event will usually fire. Removing the event listeners prevents this from causing the LocalStorage wallet name from being cleared, or errors due to the spontaneous disconnection from being caught.

Removing them explicitly is done in the React and Vue implementations as well.

Reply:
Thanks, that makes sense.